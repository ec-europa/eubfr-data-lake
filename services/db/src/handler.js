/* eslint-disable import/prefer-default-export, no-console */
export const save = (event, context, callback) => {
  /*
   * Some checks here before going any further
   */

  // Only work on the first record
  const snsRecord = event.Records[0];

  // Was the lambda triggered correctly? Is the file extension supported? etc.
  if (!snsRecord || snsRecord.EventSource !== 'aws:sns') {
    return callback('Bad record');
  }

  // Save to DB

  return callback(null, 'All fine');
};
