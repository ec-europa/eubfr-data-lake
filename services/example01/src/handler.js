/* eslint-disable import/prefer-default-export, no-console */
import getMessage from './message';

export const hello = (event, context, callback) => {
  console.log(event); // Contains incoming request data (e.g., query params, headers and more)

  const { name } = event.pathParameters || { name: 'World' };

  const response = {
    statusCode: 200,
    headers: {
      'x-custom-header': 'My Header Value',
    },
    body: JSON.stringify({ message: getMessage(name) }),
  };

  callback(null, response);
};
