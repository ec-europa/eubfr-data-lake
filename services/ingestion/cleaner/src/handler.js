/* eslint-disable import/prefer-default-export, no-console */
import AWS from 'aws-sdk'; // eslint-disable-line import/no-extraneous-dependencies

const { BUCKET } = process.env;

export const onObjectRemoved = (event, context, callback) => {
  /*
   * Some checks here before going any further
   */

  // Only work on the first record
  const snsRecord = event.Records ? event.Records[0] : undefined;

  // Was the lambda triggered correctly? Is the file extension supported? etc.
  if (!snsRecord || snsRecord.EventSource !== 'aws:sns') {
    return callback('Bad record');
  }

  /*
   * Extract information from the event
   */

  // Extract S3 record
  const s3record = JSON.parse(snsRecord.Sns.Message).Records[0];

  const s3 = new AWS.S3();

  const params = {
    Bucket: BUCKET,
    Key: `${s3record.s3.object.key}.ndjson`,
  };
  console.log('params', params);

  return s3.deleteObject(params, err => {
    if (err) {
      callback(err);
      return;
    }

    callback(null, 'object removed');
  });
};
