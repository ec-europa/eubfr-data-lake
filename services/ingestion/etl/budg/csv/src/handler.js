/* eslint-disable import/prefer-default-export, no-console */
export const parseCsv = (event, context, callback) => {
  // Contains incoming request data (e.g., query params, headers and more)
  console.log('event', event);
  console.log('context', context);

  const response = {
    statusCode: 200,
    headers: {
      'x-custom-header': 'BUDG CSV ETL',
    },
    body: JSON.stringify({ Hello: 'world!!!!!' }),
  };

  callback(null, response);
};
