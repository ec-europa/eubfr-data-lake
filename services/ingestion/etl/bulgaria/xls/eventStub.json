{
  "Records": [
    {
      "EventSource": "aws:sns",
      "EventVersion": "1.0",
      "EventSubscriptionArn": "arn:aws:sns:eu-central-1:491621799026:chernka249bg-etl-bulgaria-xlsx:2d71d46a-d632-40cc-925a-14ca4e7afc06",
      "Sns": {
        "Type": "Notification",
        "MessageId": "e9978fac-1a31-58f3-b8ad-da84e160f49a",
        "TopicArn": "arn:aws:sns:eu-central-1:491621799026:chernka249bg-etl-bulgaria-xlsx",
        "Subject": null,
        "Message": "{\"eventTime\":\"2019-04-11T08:21:36.998Z\",\"userIdentity\":{\"principalId\":\"AWS:AROAXE5XRKRZD7ULU347I:chernka249bg-storage-signed-uploads-upsert-objects\"},\"bucket\":{\"name\":\"eubfr-chernka249bg\",\"ownerIdentity\":{\"principalId\":\"ABTN0CERTFU87\"},\"arn\":\"arn:aws:s3:::eubfr-chernka249bg\"},\"object\":{\"key\":\"bulgaria/d830603f-b578-45c1-b784-736c53be06eb.xlsx\",\"size\":18813,\"eTag\":\"5f896dd6e93cfceb912162fa25cd76dc\",\"sequencer\":\"005CAEF910D7BC9A5D\"}}",
        "Timestamp": "2019-04-11T08:21:37.557Z",
        "SignatureVersion": "1",
        "Signature": "gYYwuZei4Uv9Jyd3ve11eMzUCww2C/xavnGOALJJtv+4UNUq5ZSo67NEZRyosPhpwgrpIWxDs0k+Mcnr8WcaSiLSUUniZXTMqIh6V9C2KcvzLXjJcIr8MON3cZqJaSPtKpzvBDOD3EZmpk6Crj9PJ0YvRtACXZhLlddrX/TGJSnidgRYAmxAeGgbT2hCxls6Nmt071dip3SvNvH6+W4Kr3HDSzd/M44A0ZW7Kifx6Jf/01RBN+i1xd3kncRt44NxasRYwLYurIUrePg0GVb5Wm2kmGaG8B4eRLjVvEfHSY3AKevKouT2XEDGLrdZYhr44hdyiLQPcKjz4QqroVVBpw==",
        "SigningCertUrl": "https://sns.eu-central-1.amazonaws.com/SimpleNotificationService-6aad65c2f9911b05cd53efda11f913f9.pem",
        "UnsubscribeUrl": "https://sns.eu-central-1.amazonaws.com/?Action=Unsubscribe&SubscriptionArn=arn:aws:sns:eu-central-1:491621799026:chernka249bg-etl-bulgaria-xlsx:2d71d46a-d632-40cc-925a-14ca4e7afc06",
        "MessageAttributes": {}
      }
    }
  ]
}
