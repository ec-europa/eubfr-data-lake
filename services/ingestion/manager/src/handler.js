/* eslint-disable import/prefer-default-export, no-console */
const AWS = require('aws-sdk');

export const onCreate = (event, context, callback) => {
  /*
   * Some checks here before going any further
   */

  // Only work on the first record
  const snsRecord = event.Records[0];

  // Was the lambda triggered correctly? Is the file extension supported? etc.
  if (!snsRecord || snsRecord.EventSource !== 'aws:sns') {
    return callback('Bad record');
  }

  /*
   * Extract information from the event
   */

  // Extract S3 record
  const s3record = JSON.parse(snsRecord.Sns.Message).Records[0];

  // Fill the payload
  const payload = {
    default: JSON.stringify({
      eventTime: s3record.eventTime,
      userIdentity: s3record.userIdentity,
      bucket: s3record.s3.bucket,
      object: s3record.s3.object,
    }),
  };

  /*
   * Prepare the SNS message
   */

  // Get Account ID from lambda function arn in the context
  const accountId = context.invokedFunctionArn.split(':')[4];

  // Get stage and region from environment variables
  const stage = process.env.STAGE;
  const region = process.env.REGION;

  // Get the arn
  const endpointArn = `arn:aws:sns:${region}:${accountId}:${stage}-etl-budg-csv`;

  /*
   * Send the SNS message
   */

  const sns = new AWS.SNS();

  return sns.publish(
    {
      Message: JSON.stringify(payload),
      MessageStructure: 'json',
      TargetArn: endpointArn,
    },
    err => {
      if (err) {
        console.log(err.stack);
        callback(err);
        return;
      }

      callback(null, 'push sent');
    }
  );
};
