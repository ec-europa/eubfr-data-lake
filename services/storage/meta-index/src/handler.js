/* eslint-disable import/prefer-default-export, no-console */
const AWS = require('aws-sdk'); // eslint-disable-line import/no-extraneous-dependencies

export const onObjectCreated = (event, context, callback) => {
  /*
   * Some checks here before going any further
   */

  // Only work on the first record
  const snsRecord = event.Records[0];

  // Was the lambda triggered correctly? Is the file extension supported? etc.
  if (!snsRecord || snsRecord.EventSource !== 'aws:sns') {
    return callback('Bad record');
  }

  /*
   * Extract information from the event
   */

  // Extract S3 record
  const s3record = JSON.parse(snsRecord.Sns.Message).Records[0];

  // Retrieve file meta
  const s3 = new AWS.S3();

  return s3.headObject(
    {
      Bucket: s3record.s3.bucket.name,
      Key: s3record.s3.object.key,
    },
    (err, data) => {
      if (err) return callback(err); // an error occurred

      console.log({
        Bucket: s3record.s3.bucket.name,
        Key: s3record.s3.object.key,
      });
      console.log(JSON.stringify(data));
      return callback(null, 'All fine');
    }
  );

  // Save record
  // dynamodb
};
