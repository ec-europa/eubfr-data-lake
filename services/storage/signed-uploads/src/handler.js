/* eslint-disable import/prefer-default-export */

import AWS from 'aws-sdk'; // eslint-disable-line import/no-extraneous-dependencies

const bucket = process.env.BUCKET;
const region = process.env.REGION;

const s3 = new AWS.S3({ signatureVersion: 'v4', region });

export const onProjectsUpsert = (event, context, callback) => {
  const file =
    event.headers && event.headers['x-amz-meta-producer-key']
      ? event.headers['x-amz-meta-producer-key']
      : undefined;

  if (!file) {
    const response = {
      statusCode: 400,
      body: JSON.stringify({
        message: `Missing x-amz-meta-producer-key header`,
      }),
    };

    callback(null, response);
  }

  // If producer has correctly submitted a key.
  const params = {
    Bucket: bucket,
    Key: file,
    Expires: 30,
  };

  s3.getSignedUrl('putObject', params, (err, url) => {
    if (err) {
      callback(err);
    }

    const response = {
      statusCode: 200,
      headers: {
        'Access-Control-Allow-Origin': '*', // Required for CORS support to work
        'Access-Control-Allow-Credentials': true, // Required for cookies, authorization headers with HTTPS
      },
      body: JSON.stringify(url),
    };

    callback(null, response);
  });
};

export const onProjectsGetRoot = (event, context, callback) => {
  const response = {
    statusCode: 200,
    headers: {
      'Access-Control-Allow-Origin': '*', // Required for CORS support
      'Access-Control-Allow-Credentials': true, // Required for cookies, authorization headers with HTTPS
    },
    body: JSON.stringify({
      message:
        'Docroot is not used at the moment. Please check the other endpoints',
    }),
  };

  callback(null, response);
};
