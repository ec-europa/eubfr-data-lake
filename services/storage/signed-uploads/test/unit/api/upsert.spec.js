/**
 * @jest-environment node
 */

import AWS from 'aws-sdk-mock';
import AWS_SDK from 'aws-sdk';

import handler from '../../../src/api/upsert'; // eslint-disable-line import/no-named-as-default
import eventStub from '../../stubs/eventHttpApiGateway.json';

// Explicitly set the correct module, as it might not map correctly after transpilation.
AWS.setSDKInstance(AWS_SDK);

describe(`Service aws-node-singned-uploads: S3 mock for successful operations`, () => {
  beforeAll(() => {
    AWS.mock('S3', 'getSignedUrl', (method, _, callback) => {
      callback(null, {
        data: 'https://example.com',
      });
    });

    AWS.mock('IAM', 'listGroupsForUser', (method, callback) => {
      callback(null, {
        Groups: [
          {
            GroupName: 'Producers',
          },
        ],
      });
    });
  });

  afterEach(() => {
    delete process.env.BUCKET;
    delete process.env.REGION;
  });

  afterAll(() => {
    AWS.restore('S3');
    AWS.restore('IAM');
  });

  test('Require environment variables', async () => {
    try {
      await handler();
    } catch (error) {
      expect(error.message).toEqual(
        'BUCKET and REGION environment variables are required!'
      );
    }
  });

  test('Require a header "x-amz-meta-producer-key"', async () => {
    process.env.BUCKET = 'foo';
    process.env.REGION = 'bar';

    const event = {
      requestContext: {
        identity: {
          userArn: 'arn:aws:iam::123456789012:user/test',
        },
      },
    };

    const context = {};

    try {
      const result = await handler(event, context);
      expect(result).toMatchSnapshot();
    } catch (error) {
      console.error(error);
    }
  });

  test('Replies back with a JSON for a signed upload on success', async () => {
    process.env.BUCKET = 'foo';
    process.env.REGION = 'bar';

    // Deep clone
    const event = JSON.parse(JSON.stringify(eventStub));
    const context = {};

    try {
      const result = await handler(event, context);
      expect(result).toMatchSnapshot();
    } catch (error) {
      console.error(error);
    }
  });
});

describe('Service aws-node-singned-uploads: S3 mock for failed operations', () => {
  beforeAll(() => {
    AWS.mock('S3', 'getSignedUrl', (method, _, callback) => {
      callback('S3 failed');
    });

    AWS.mock('IAM', 'listGroupsForUser', (method, callback) => {
      callback(null, {
        Groups: [
          {
            GroupName: 'Producers',
          },
        ],
      });
    });
  });

  afterEach(() => {
    delete process.env.BUCKET;
    delete process.env.REGION;
  });

  afterAll(() => {
    AWS.restore('S3');
    AWS.restore('IAM');
  });
});
