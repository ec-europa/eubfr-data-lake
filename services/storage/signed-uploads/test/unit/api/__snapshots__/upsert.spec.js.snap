// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Service aws-node-singned-uploads: S3 mock for successful operations Replies back with a JSON for a signed upload on success 1`] = `
Object {
  "body": "{}",
  "headers": Object {
    "Access-Control-Allow-Credentials": true,
    "Access-Control-Allow-Origin": "*",
  },
  "statusCode": 200,
}
`;

exports[`Service aws-node-singned-uploads: S3 mock for successful operations Require a header "x-amz-meta-producer-key" 1`] = `
Object {
  "body": "{\\"message\\":\\"Missing x-amz-meta-producer-key header\\"}",
  "statusCode": 400,
}
`;
